QHyper/problems/Knapsack.py

```diff
@@ -0,0 +1,131 @@
+import random
+from collections import namedtuple
+
+from .problem import Problem
+
+ 
+Item = namedtuple('Item', "weight value")
+
+
+class Knapsack:
+    """Knapsack class"""
+
+    def __init__(self, max_weight: int, N: int=0, max_item_value: int=10) -> None:
+        self.items: list[Item] = []
+        self.all_items: int = N
+        self.max_weight: int = max_weight
+        self.max_item_value: int = max_item_value
+        self.generate_knapsack(N)
+
+    def generate_knapsack(self, N: int) -> None:
+        for _ in range(N):
+            self.items.append(Item(
+                    random.randint(1, self.max_weight), 
+                    random.randint(1, self.max_item_value)
+                )
+            )
+
+    def set_knapsack(self, items: list[tuple[int, int]]) -> None:
+        self.items = [Item(weight, value) for weight, value in items]
+        self.all_items = len(items)
+
+
+class KnapsackProblem(Problem):
+    """Class defining objective function and constraints for knapsack problem
+    
+    Attributes
+    ----------
+    objective_function : str
+        objective function in SymPy syntax
+    constraints : list[str]
+        list of constraints in SymPy syntax
+    wires : int
+        number of qubits in the circuit, equals to number of items in knapsack + max weight of knapsack
+    """
+
+    def __init__(
+        self, 
+        knapsack: Knapsack, 
+        # number_of_layers: int = 6, 
```

Comment:
This options belong to QAOA solver, should not be here.

Reply:
u mean `number_of_layers`?